### ISSUE TITLE
Staging Review finding: Filter and sort outcomes are unclear for screen reader users

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#93695]
_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

## Product Information
**Team:** Transition Experience
**Product:** Transition Experience MVP v1
**Feature:** V1 Transition Experience

## Findings details
**VA.gov Experience Standard - issue:** [User experiences an unexpected interaction with a component or pattern.](https://design.va.gov/about/experience-standards/#user-experiences-predictable-interactions-with-components)
**VA.gov Experience Standard - category:** [Usability](https://design.va.gov/about/experience-standards/#category-usability)
**Launch-blocking:** No
**Design System review:** No
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

When filtering or sorting on the results page, the results change immediately when changing the value of the `select` dropdown. For screen reader users, the new value of the `select` is announced, but if you can't see the visual cues of content on the page changing then there's no indication that the action has actually taken place.

<details><summary>Link, screenshot or steps to recreate</summary>

![Filter and sort select dropdowns, which have no accompanying button to trigger their actions.](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/working-with-vsp/vsp-collaboration-cycle/staging-review-images/82863-Brian-4.png?raw=true)

</details>

## Recommended action

There's a lot of active discussion around filter/sort interactions on VA.gov and an effort to create a new pattern that meets accessibility requirements while being intuitive to all users. I'm noting that here because guidance may change in the not-too-distant future --- but I'll make this recommendation in the meantime, which is hopefully not too far off from where this ultimately lands. (cc @rtwell for awareness.)

- Page content should not change on input --- especially not for filtering, as a screen reader user may accidentally lose relevant information when they thought they were just exploring their options in the `select`.
- Instead, there should be a button (maybe "Filter and sort") that follows the `select` dropdowns, and that button should be what triggers the changes to page content.
- When you press that button, keyboard focus should move to the summary text "Showing X results, filtered to show Y, sorted Z," using `tabindex="-1"` to make that text focusable.

Moving focus in this way ensures that screen reader users get an immediate success message for their action and an immediate summary of the current state of the page. And doing it when pressing the button rather than on focus ensures that they don't get bounced to this new place before they're ready, by making the filter/sort action a deliberate choice.

## References

- **Accessibility Defect Severity:** [2: High. Should be fixed in 1-2 sprints post-launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-2)
- **WCAG Success Criteria:** [WCAG 3.2.2 On Input - Level A](https://www.w3.org/TR/WCAG21/#on-input)
- **Modality:** Screen reader
- **Design System Component:** N/A
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A

---

### Next Steps for the VFS Team

- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue, comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by nkontrabecki** [comment]
Confirmed fixed and tested in Sprint 15.
